### PR TITLE
feat(pd): integrate `pd-core` into hugegraph

### PR DESCRIPTION
**Pull request by VGalaxies**
subtask of #2265

The corresponding tests will be merged after `pd-service`

For detailed module analysis documentation, please refer to [this link](https://hugegraph.feishu.cn/wiki/GcjGwygZziNbHgkMRA6ctnWUnsf).

### CODE REVIEW

**Comment by VGalaxies** [comment]
@imbajin Do we need to address the Code scanning results in this PR?

**Comment by imbajin** [comment]
> @imbajin Do we need to address the Code scanning issues in this PR?

miss the context, any details?

**Comment by VGalaxies** [comment]
> > @imbajin Do we need to address the Code scanning issues in this PR?
> 
> miss the context, any details?

https://github.com/apache/incubator-hugegraph/pull/2478/checks?check_run_id=22664435037

File: hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/TaskScheduleService.java [hunk 1]
```diff
@@ -0,0 +1,845 @@
...
+
+/**
+ * 任务调度服务，定时检查Store、资源、分区的状态，及时迁移数据，错误节点
+ * 1、监测Store是否离线
+ * 2、监测Partition的副本是否正确
+ * 3、监测Partition的工作模式是否正确
+ * 4、监测Partition是否需要分裂,监测分裂是否完成
+ */
+@Slf4j
+public class TaskScheduleService {
```
Comment (javeme) [inline hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/TaskScheduleService.java:54]: seems it's not a service, prefer to  keep TaskScheduler name, and also move to a package 
 like pd.task

File: hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/ShardGroupStatusListener.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hugegraph.pd;
+
+import org.apache.hugegraph.pd.grpc.Metapb;
+
+public interface ShardGroupStatusListener {
```
Comment (javeme) [inline hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/ShardGroupStatusListener.java:22]: can we move all the Listener files to package pd.listener, and move all the Service files to package pd.service

File: hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/TaskScheduleService.java [hunk 3]
```diff
@@ -0,0 +1,845 @@
...
+        movedPartitions.forEach((partId, storePair) -> {
+            // 源和目标storeID都不为0
+            if (storePair.getKey() > 0 && storePair.getValue() > 0) {
+                partitionService.movePartitionsShard(partId, storePair.getKey(),
+                                                     storePair.getValue());
+            }
+        });
+        return movedPartitions;
+    }
+
```
Comment (javeme) [inline hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/TaskScheduleService.java:844]: useless blank line

**Comment by javeme** [comment]
One more suggestion, we can split the files under the pd client into multiple directories according to functions, and provide Clients.java for navigation(like `Clients.createXxClient()`). In addition, the files naming of pd proto need to be classified/formatted.

**Comment by imbajin** [comment]
> One more suggestion, we can split the files under the pd client into multiple directories according to functions, and provide Clients.java for navigation(like `Clients.createXxClient()`). In addition, the files naming of pd proto need to be classified/formatted.

@javeme @zyxxoo 

The **code-style** related issues(name/comment/format/clean) may list first(although we already cleaned a lot of them..), the style problems are better to be **modified together** after reaching TODOs to avoid more **inconsistencies** under version merge. 
For now, we can first focus on the problems in **code logic** and **implementation**, which can only be considered as a tradeoff 😿  (so as the PR #2272 )

**Comment by VGalaxies** [comment]
merge directly into the master branch after https://github.com/apache/incubator-hugegraph/pull/2498...

File: hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/config/PDConfig.java [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+         * [1-9][ ](second, minute, hour, day, month, year)
+         * unit could not be null, the number part is 1 by default.
+         *
+         * @param exp
+         * @return seconds value of the expression. 1 will return by illegal expression
+         */
+        private Long parseTimeExpression(String exp) {
+            if (exp != null) {
+                Pattern pattern = Pattern.compile(
+                        "(?<n>(\\d+)*)(\\s)*(?<unit>(second|minute|hour|day|month|year)$)");
```
[the remark this answers is not in the document]
Comment (VGalaxies) [reply hugegraph-pd/hg-pd-core/src/main/java/org/apache/hugegraph/pd/config/PDConfig.java:206]: ignore it for now...

**Comment by VGalaxies** [comment]
@imbajin All Code scanning warning have been fixed except for this one:

image

**Review by imbajin — APPROVED** [review]
THX ❤️~

**Review by liuxiaocs7 — APPROVED** [review]

**Pull request merged**